src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets

```diff
@@ -61,6 +61,42 @@ Copyright (c) .NET Foundation. All rights reserved.
     <RuntimeIdentifier Condition="'$(PlatformTarget)' == 'x86' or '$(PlatformTarget)' == ''">win7-x86</RuntimeIdentifier>
   </PropertyGroup>
 
+  <!-- Check if SelfContained is global, resets the value to it's original state if it was local.-->
+  <PropertyGroup>
+    <_OriginalSelfContainedValue>$(SelfContained)</_OriginalSelfContainedValue>
+    <SelfContained></SelfContained>
+  </PropertyGroup>
+
+  <Choose>
+    <!-- Assumes a global variable cannot be an undefined variable.-->
+    <When Condition="'$(SelfContained)' != '' and $(SelfContained) == $(_OriginalSelfContainedValue)">
+      <PropertyGroup>
+        <_SelfContainedIsGlobal>true</_SelfContainedIsGlobal>
+      </PropertyGroup>
+    </When>
+    <Otherwise>
+      <PropertyGroup>
+        <_SelfContainedIsGlobal>false</_SelfContainedIsGlobal>
+      </PropertyGroup>
+    </Otherwise>
+  </Choose>
+
+  <PropertyGroup>
+    <SelfContained>$(_OriginalSelfContainedValue)</SelfContained>
+    <_OriginalSelfContainedValue></_OriginalSelfContainedValue>
+  </PropertyGroup>
```

Comment:
Any reason to use a `Choose` block instead of something like this?

```suggestion
  <PropertyGroup>
    <_OriginalSelfContainedValue>$(SelfContained)</_OriginalSelfContainedValue>
    <SelfContained></SelfContained>

    <_SelfContainedIsGlobal>false</_SelfContainedIsGlobal>
    <!-- Assumes a global variable cannot be an undefined variable.-->
    <_SelfContainedIsGlobal Condition="'$(SelfContained)' != '' and $(SelfContained) == $(_OriginalSelfContainedValue)">true</_SelfContainedIsGlobal>

    <SelfContained>$(_OriginalSelfContainedValue)</SelfContained>
    <_OriginalSelfContainedValue></_OriginalSelfContainedValue>
  </PropertyGroup>
```

Reply:
Thanks, this is helpful for my learning... Trying to think of MSBuild in if/elses is not the best idea it seems